### ISSUE TITLE
Builds fail with new black 24.1.0 linting rules.

### ISSUE THREAD

**Issue by keraion** [sqlfluff/sqlfluff#5572]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running `tox -e linting` the new stable rules for black 24.1.0 fail.

### Expected Behaviour

Linting should be successful.

### Observed Behaviour

```sh
linting: commands[1]> flake8
./plugins/sqlfluff-plugin-example/setup.py:2:1: BLK100 Black would make changes.
./plugins/sqlfluff-plugin-example/test/rules/rule_test_cases_test.py:2:1: BLK100 Black would make changes.
...
```

### How to reproduce

Run `tox -e linting`

### Dialect

N/A

### Version

main branch

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
